demography.md

```diff
@@ -127,101 +170,223 @@ population 0 than vice versa.
 
 ```{code-cell} ipython3
-migration_rate_change = msprime.MigrationRateChange(time = 100, rate = 0.01, source=0, dest=1)
+dem.add_migration_rate_change(time=100, rate=0.01, source=0, dest=1)
+dem
 ```
 
-A list of these changes can be supplied to {func}`msprime.sim_ancestry` via the
-``demographic_events`` input:
-(If there is more than 1 change, ensure they are ordered by backwards-time!)
+The output above shows that we have successfully added our first demographic event to our {class}`msprime.Demography` object, a migration rate change.
+We are now ready to simulate:
 
 ```{code-cell} ipython3
-dem = msprime.Demography(populations=[pop0, pop1], migration_matrix=M, events=[migration_rate_change])
-ts = msprime.sim_ancestry(samples=[samples0, samples1], demography=dem, sequence_length=1000, random_seed=63461, recombination_rate=1e-7)
+ts = msprime.sim_ancestry(
+  samples={"Population0" : 3, "Population1" : 3},
+  demography=dem,
+  sequence_length=1000,
+  random_seed=63461,
+  recombination_rate=1e-7)
+ts
 ```
+### Admixture and population splits
 
-### Mass migrations
+#### Admixture
 
-:class:`msprime.MassMigration` objects are used to specify one-off events in which some fraction of a population moves into another population. These are useful for specifying divergence and admixture events.
+It is also easy to specify admixture and divergence events with msprime.
+Suppose we wanted to specify our demography so that 50 generations ago,
+30% of population 0 was a migrant from population 1:
 
-![](_static/tute-mass-migration.png)
+```{todo}
+make a diagram showing this
+```
 
-You'll need to provide the time of the event in generations,
-the ID of the source and destination populations,
-and a migration proportion (which defaults to 1.0).
-For example, the following specifies that 50 generations ago,
-30% of population 0 was a migrant from population 1.
+We can do this by using the {meth}`msprime.Demography.add_admixture` method on our demography object.
+We must supply a list of ancestral populations participating in the admixture, and a list of the same size specifying the proportions of migrants from each of these populations.
 
 ```{code-cell} ipython3
-admixture_event  = msprime.MassMigration(time = 50, source = 0, dest = 1, proportion = 0.3)
+dem = msprime.Demography()
+dem.add_population(
+  name="AncestralPop0",
+  description="Plotted in red.",
+  initial_size=500,
+  growth_rate=0
+  )
+dem.add_population(
+  name="AncestralPop1",
+  description="Plotted in blue.",
+  initial_size=500,
+  growth_rate=0
+  )
+dem.add_population(
+  name="AdmixedPop",
+  description="Plotted in green.",
+  initial_size=500,
+  growth_rate=0
+  )
+dem.set_migration_rate(source=0, dest=1, rate=0.05)
+dem.set_migration_rate(source=1, dest=0, rate=0.02)
+
+# Specify admixture event.
+dem.add_admixture(
+  time=50,
+  derived="AdmixedPop",
+  ancestral=["AncestralPop0", "AncestralPop1"],
+  proportions=[0.3, 0.7])
```

Comment:
Put a semicolon at the end here to prevent the Admixture event that is returned from being rendered by Jupyter.